### ISSUE TITLE
Support Puppeteer-Firefox

### ISSUE THREAD

**Issue by chanjsq** [berstend/puppeteer-extra#55]
Is it possible to support Puppeteer-Firefox? Thanks.

**Comment by berstend** [comment]
Hey there, I'll neglected this project a bit and will do some hygiene first (updating deps, etc) and refactor the core to TS. Afterwards I'll look into the effort needed to support FF as well :)

**Comment by berstend** [comment]
This is a good overview about what's already supported:
https://aslushnikov.github.io/ispuppeteerfirefoxready/

Looks pretty decent, most plugins should theoretically work with FF as well.

**Comment by chanjsq** [comment]
awesome 👍

**Comment by berstend** [comment]
@chanjsq I've added `addExtra` support while rewriting the core in TypeScript, this is now supported (I just published `v3.1.3`):

```js
const { addExtra } = require('puppeteer-extra')
const puppeteer = addExtra(require('puppeteer-firefox'))

puppeteer.launch({ headless: false }).then(async browser => {
  const page = await browser.newPage()
  await page.setViewport({ width: 800, height: 600 })

  await page.goto('https://www.spacejam.com/archive/spacejam/movie/jam.htm')
  await page.waitFor(10 * 1000)

  await browser.close()
})
```
